### ISSUE TITLE
Why ignore bool type when uses WidthAttribute?

### ISSUE THREAD

**Issue by raslab** [slavniyteo/one-line#40]
Today i tried to add `WidthAttribute` to my bool field in table created with `OneLineWithHeader` and catch some strange result (image like example):
![Screenshot from 2019-09-26 00-54-32](https://user-images.githubusercontent.com/2966314/65643053-62dcd080-dff9-11e9-9ae8-2d82cfd7c766.png)

After this started ti research error and show
https://github.com/slavniyteo/one-line/blob/7f0bf8bf496e75a3f232ab37ec2778d4c0e46be6/Editor/Scripts/Drawers/SimpleFieldDrawer.cs#L26-L30

Why?! :(

**Comment by slavniyteo** [comment]
You are right, it seems wierd. 

It happens because of all fields have attribute `Width` (and `Width` sets weight to zero by default). If at least one of them doesn't have `Width` or has `Width` and non-zero `Weight`, i'll fill all free space on the line. In most cases you are suggested to follow this *at-least-one-weighted-field* way. 

```csharp
[Width(100), Weight(1)] public String ItemId; //Use Weight and Width
[Weight(1)] public String ItemName; //Use Weight only; Weight=1 is default when Width is not present.
```

But in any case, flexibility is preferred. It would be better to search for attributes and if nothing is found, then use default values. The same applies to `GetWeight` method.

I mark this issue as a bug and will fix it next time I'll have time on my hands.

**Comment by raslab** [comment]
I'm sorry, looks like i need to improve my bug reporting skills :)
As for me, fixed width isn't problem, main problem then a can't see labels on bool fields when i using `OneLineWithHeader`)

![image](https://user-images.githubusercontent.com/2966314/65695176-b5f96680-e07f-11e9-8d84-d3ff291fe26a.png)

How solution can be figured out? Add some "force" flag to `WeightAttribute` constructor or maybe pass bool fields with default rules if `OneLineWithHeader` uses?

**Comment by slavniyteo** [comment]
To allow `Width` on boolean fields, modify these two methods:

```csharp
public float GetWeight(SerializedProperty property){
    var attributes = property.GetCustomAttributes<WeightAttribute>();

    if (attributes.Any()) {
        return attributes.Select(x => x.Weight)
                         .Sum();
    } 
    else {
        switch (property.propertyType){
            case SerializedPropertyType.Boolean: {
                return 0;
            }
            default: {
                return 1;
            }
        }
    }
}

public float GetFixedWidth(SerializedProperty property){
    var attributes = property.GetCustomAttributes<WidthAttribute>();

    if (attributes.Any()) {
        return attributes.Select(x => x.Width)
                         .Sum();
    } 
    else {
        switch (property.propertyType){
            case SerializedPropertyType.Boolean: {
                return EditorGUIUtility.singleLineHeight - 2;
            }
            default: {
                return 0;
            }
        }
    }
}
```

I write it in browser and can't check it with spellchecker or compiler, but my code represents the main idea. Please try it.

---

The problem of `OneLineHeader` with boolean fields is too long field names and too short field representation (a small checkbox). That's why I hardcoded boolean fields width. But if you really want to specify it's width, you should be able to do it.

**Comment by raslab** [comment]
Good idea, thanks, will try at evening

**Comment by raslab** [comment]
Yee, it's works, thanks :)
![Screenshot from 2019-09-28 01-41-24](https://user-images.githubusercontent.com/2966314/65806263-5d19f300-e191-11e9-9be5-167d2788c8d0.png)

P.S.: also prepared pull request to make your life easier )

**Comment by slavniyteo** [comment]
Fixed with #41
